src/module/rolls/roll-graphic.ts

```diff
@@ -14,14 +14,16 @@ type RenderData = {
   adds: SourcedValue<number | string>[]
   computedActionDie?: SourcedValue<number | string>
   actionIsDie?: boolean
-  actionMinMax?: string
+  actionMinMax?: 'min' | 'max' | undefined
   actionScore?: number
   actionScoreCapped?: boolean
   actionDieCanceledByNegativeMomentum?: boolean
 
-  challengeDice: Partial<SourcedValue & { minmax?: string }>[]
+  challengeDice: Partial<
+    SourcedValue & { minmax?: 'min' | 'max' | undefined }
```

Comment:
Nit: the `?:` here includes the `| undefined` part of the type.

Reply:
oh, ive been meaning to bring that up, actually - most TS best practices i've seen recommend using [strict null checks](https://www.typescriptlang.org/tsconfig#strictNullChecks). we don't currently have that flag set, but maybe it'd be a good idea?

Reply:
Separate PR, my friend. 😀